### ISSUE TITLE
ERROR: Failure in AnalysisManager.run

### ISSUE THREAD

**Issue by focaaby** [cuckoosandbox/cuckoo#1133]
Hello

I submitted a file to Cuckoo , and got this following error:

Environment : 
- MacOS : 10.12
- Cuckoo version : 2.0-rc1

```

                                 _|
     _|_|_|  _|    _|    _|_|_|  _|  _|      _|_|      _|_|
   _|        _|    _|  _|        _|_|      _|    _|  _|    _|
   _|        _|    _|  _|        _|  _|    _|    _|  _|    _|
     _|_|_|    _|_|_|    _|_|_|  _|    _|    _|_|      _|_|

 Cuckoo Sandbox 2.0-rc1
 www.cuckoosandbox.org
 Copyright (c) 2010-2015

 Checking for updates...
 Good! You have the latest version available.

2016-10-20 02:04:56,561 [root] INFO: Updated running task ID 4 status to failed_analysis
2016-10-20 02:04:56,573 [lib.cuckoo.core.scheduler] INFO: Using "avd" as machine manager
2016-10-20 02:04:56,597 [lib.cuckoo.core.scheduler] INFO: Loaded 1 machine/s
2016-10-20 02:04:56,618 [lib.cuckoo.core.scheduler] INFO: Unable to determine the network interface for VM with name aosx_1, Cuckoo will not be able to give it full internet access or route it through a VPN! Please define a default network interface for the machinery or define a network interface for each VM.
2016-10-20 02:04:56,619 [lib.cuckoo.core.scheduler] INFO: Waiting for analysis tasks.
2016-10-20 02:05:18,171 [lib.cuckoo.core.scheduler] INFO: Starting analysis of FILE "CCC01FD6D875B95E2AF5F270AAF8E842.576B9B86.apk" (task #5, options "")
2016-10-20 02:05:18,180 [lib.cuckoo.core.scheduler] INFO: File already exists at "/Users/focaaby/Project/rootguard/cuckoo/storage/binaries/36f03d7a2e15cf3ee8f4e0f8bc876a1f80a99ad0eacb398f5272e179c60ed4f8"
2016-10-20 02:05:18,193 [lib.cuckoo.core.scheduler] INFO: Task #5: acquired machine aosx_1 (label=aosx_1)
2016-10-20 02:05:54,455 [lib.cuckoo.core.guest] INFO: Starting analysis on guest (id=aosx_1, ip=127.0.0.1)
2016-10-20 02:05:54,484 [modules.machinery.avd] INFO: Stopping AVD listening on port 5554
2016-10-20 02:05:55,569 [lib.cuckoo.core.scheduler] ERROR: Failure in AnalysisManager.run
Traceback (most recent call last):
  File "/Users/focaaby/Project/rootguard/cuckoo/lib/cuckoo/core/scheduler.py", line 447, in run
    self.launch_analysis()
  File "/Users/focaaby/Project/rootguard/cuckoo/lib/cuckoo/core/scheduler.py", line 339, in launch_analysis
    self.guest_manage(options)
  File "/Users/focaaby/Project/rootguard/cuckoo/lib/cuckoo/core/scheduler.py", line 257, in guest_manage
    guest.start_analysis(options, monitor)
  File "/Users/focaaby/Project/rootguard/cuckoo/lib/cuckoo/core/guest.py", line 382, in start_analysis
    r = self.get("/")
  File "/Users/focaaby/Project/rootguard/cuckoo/lib/cuckoo/core/guest.py", line 255, in get
    return requests.get(url, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 69, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 50, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
ConnectionError: ('Connection aborted.', BadStatusLine("''",))
```

**Comment by doomedraven** [comment]
try dev version, it have a lot of this issues fixed

**Comment by focaaby** [comment]
Hello @doomedraven 
I have git clone Cuckoo repo, but still faced the same problem.

```

    sSSs   .S       S.     sSSs   .S    S.     sSSs_sSSs      sSSs_sSSs
   d%%SP  .SS       SS.   d%%SP  .SS    SS.   d%%SP~YS%%b    d%%SP~YS%%b
  d%S'    S%S       S%S  d%S'    S%S    S&S  d%S'     `S%b  d%S'     `S%b
  S%S     S%S       S%S  S%S     S%S    d*S  S%S       S%S  S%S       S%S
  S&S     S&S       S&S  S&S     S&S   .S*S  S&S       S&S  S&S       S&S
  S&S     S&S       S&S  S&S     S&S_sdSSS   S&S       S&S  S&S       S&S
  S&S     S&S       S&S  S&S     S&S~YSSY%b  S&S       S&S  S&S       S&S
  S&S     S&S       S&S  S&S     S&S    `S%  S&S       S&S  S&S       S&S
  S*b     S*b       d*S  S*b     S*S     S%  S*b       d*S  S*b       d*S
  S*S.    S*S.     .S*S  S*S.    S*S     S&  S*S.     .S*S  S*S.     .S*S
   SSSbs   SSSbs_sdSSS    SSSbs  S*S     S&   SSSbs_sdSSS    SSSbs_sdSSS
    YSSP    YSSP~YSSY      YSSP  S*S     SS    YSSP~YSSY      YSSP~YSSY
                                 SP
                                 Y

 Cuckoo Sandbox 2.0-dev
 www.cuckoosandbox.org
 Copyright (c) 2010-2015

 Checking for updates...
 You are running a development version! Current stable is 2.0-rc1.
2016-10-20 10:51:56,680 [lib.cuckoo.core.resultserver] WARNING: Cannot bind ResultServer on port 2042, trying another port.
2016-10-20 10:51:56,681 [lib.cuckoo.core.scheduler] INFO: Using "avd" as machine manager
2016-10-20 10:51:56,771 [lib.cuckoo.core.scheduler] INFO: Loaded 1 machine/s
2016-10-20 10:51:56,792 [lib.cuckoo.core.scheduler] INFO: Unable to determine the network interface for VM with name aosx_1, Cuckoo will not be able to give it full internet access or route it through a VPN! Please define a default network interface for the machinery or define a network interface for each VM.
2016-10-20 10:51:56,793 [lib.cuckoo.core.scheduler] INFO: Waiting for analysis tasks.
2016-10-20 10:52:19,356 [lib.cuckoo.core.scheduler] INFO: Starting analysis of FILE "CCC01FD6D875B95E2AF5F270AAF8E842.576B9B86.apk" (task #1, options "")
2016-10-20 10:52:19,386 [lib.cuckoo.core.scheduler] INFO: Task #1: acquired machine aosx_1 (label=aosx_1)
2016-10-20 10:53:16,351 [lib.cuckoo.core.guest] INFO: Starting analysis on guest (id=aosx_1, ip=127.0.0.1)
2016-10-20 10:53:16,369 [modules.machinery.avd] INFO: Stopping AVD listening on port 5554
2016-10-20 10:53:17,466 [lib.cuckoo.core.scheduler] ERROR: Failure in AnalysisManager.run
Traceback (most recent call last):
  File "/Users/focaaby/Project/cuckoo/lib/cuckoo/core/scheduler.py", line 519, in run
    self.launch_analysis()
  File "/Users/focaaby/Project/cuckoo/lib/cuckoo/core/scheduler.py", line 399, in launch_analysis
    self.guest_manage(options)
  File "/Users/focaaby/Project/cuckoo/lib/cuckoo/core/scheduler.py", line 301, in guest_manage
    self.guest_manager.start_analysis(options, monitor)
  File "/Users/focaaby/Project/cuckoo/lib/cuckoo/core/guest.py", line 384, in start_analysis
    r = self.get("/")
  File "/Users/focaaby/Project/cuckoo/lib/cuckoo/core/guest.py", line 290, in get
    return session.get(url, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 477, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
ConnectionError: ('Connection aborted.', BadStatusLine("''",))
```

**Comment by doomedraven** [comment]
it looks like it can't connect to your vm, start you vm manually and try `curl vm_io:8000` if you will get 50x error, then it works, if not then your agent not listening

**Comment by focaaby** [comment]
My vm is AVD, so how could I install my agent in my vm ? 
and I have already installed [these tools](http://cuckoo-droid.readthedocs.io/en/latest/installation/guest_android_avd/requirements/) in vm.

**Comment by jbremer** [comment]
@focaaby Did you have any progress on this issue? It definitely does look like Cuckoo is not able to reach the Agent in your AVD. Did you setup the proper IP addresses etc?

**Comment by jbremer** [comment]
Any update here?

**Comment by blacvortex** [comment]
I am running into the same error. In my case the agent responds back with a Guest is running Cuckoo Agent 0.5. This happens when the analyzer is transferred.

**Comment by christopherferreira3** [comment]
Did you managed to make any progress on this? I'm experiencing the same problem...

**Comment by jbremer** [comment]
We've certainly made some progress in this area. Error reporting should now give more information etc. @christopherferreira3 please provide us more information about your setup etc.

**Comment by christopherferreira3** [comment]
Hi @jbremer ! I managed to fix this issue but unfortunately, I don't remember how. All I remember is that the error was on my side.

The only problem I'm having now is that I run into a  critical timeout (during status=2). The report comes with tons of information except Droidmon was empty. Am I doing anything wrong? I've also checked with logcat and I'm able to see some action from Droidmon.
